### PR TITLE
feat: resolve types across i/o-mappings

### LINKED ISSUE #2: Resolve types across i/o-mappings
__Is your feature request related to a problem? Please describe__

- When I defined output data and filter it using output mappings, I want the new variable to have the same type as the variable I set

__Describe the solution you'd like__

- Feel expressions are evaluated and the type and description is kept

__Describe alternatives you've considered__

-

__Additional context__

related to https://github.com/bpmn-io/internal-docs/issues/680

### PR DESCRIPTION
**Pull request by marstamm**
closes #2

### CODE REVIEW

**Comment by marstamm** [comment]
After a discussion with @barmac, these are improvements that can be made:
- Do not override private methods, offer an API for extension instead
- Is it necessary for `parseIoMappings` to override the existing variable object? Maybe it is possible to make it side-effect free and create a new variable object

**Comment by barmac** [comment]
@marstamm Is this ready for review?

**Comment by marstamm** [comment]
Yes, the changes we discussed on Friday are implemented in [4f225e9](https://github.com/bpmn-io/variable-resolver/pull/5/commits/4f225e9c7ef8f50ec7cfc9dba0893a3001758093) and [7222f40](https://github.com/bpmn-io/variable-resolver/pull/5/commits/7222f4068556fce48a7edccb9e6dd7492d4f9eb1)

**Comment by barmac** [comment]
Cool, I will check this out at most tomorrow!

File: lib/ZeebeVariableResolver.js [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+import { parseIoMappings } from './util/feelUtility';
+
+const HIGH_PRIORITY = 2000;
 
 export class ZeebeVariableResolver extends BaseVariableResolver {
   constructor(eventBus, bpmnjs) {
     super(eventBus, bpmnjs);
     this._baseExtractor = getProcessVariables;
+
+    eventBus.on('variableResolver.parseVariables', (e, context) => {
```
Comment (barmac) [inline lib/ZeebeVariableResolver.js:12]: The listener is registered twice: once here, and once in the base variable resolver. Is this intended?

Comment (marstamm) [reply lib/ZeebeVariableResolver.js:12]: Yes, this is intended. We still want the behavior of the `baseVariableResolver`, but extend it by parsing the IO mappings first. This is why this one one is registered with `HIGH_PRIORITY`

**Review by barmac — APPROVED** [review]
Merging this but pls add documentation for the base variable resolver.

**Pull request merged**